tests/unit/dcc_comms/test_mqtt_dcc_comms.py

```diff
@@ -0,0 +1,170 @@
+
+class MqttDccCommsTest(unittest.TestCase):
+    """
+    Unit test cases for MQTT DccComms
+    """
+
+    def setUp(self):
+        """
+        Setup all required parameters for MQTT DCC communication tests
+        :return:
+        """
+        # Monkey patched MQTT connect method
+        Mqtt.connect_soc = mock_connect_soc
+
+        # Create Edge System
+        self.edge_system = Dell5KEdgeSystem(config['EdgeSystemName'])
+
+        self.url = config['BrokerIP']
+        self.port = config['BrokerPort']
+        self.clean_session = True
+        self.publish_message = "test-message"
+
+        # Creating messaging attributes
+        self.mqtt_msg_attr = MqttMessagingAttributes(pub_topic="publish-topic")
+
+        # Instantiate client for DCC communication
+        self.client = MqttDccComms(edge_system_name=self.edge_system.name,
+                                   url=self.url, port=self.port, clean_session=self.clean_session)
+
+    def test_init_no_msgattr(self):
+        """
+        Test MqttDccComms initialisation with:
+        #1. Empty client ID
+        #2. Non-empty client ID
+        :return:
+        """
+        Mqtt.connect_soc = mock_connect_soc
+
+        # Test with empty client ID
+        self.assertIsNotNone(MqttDccComms(edge_system_name=self.edge_system.name,
+                                          url=self.url, port=self.port,
+                                          clean_session=self.clean_session))
+
+        # Test with non-empty client ID
+        self.assertIsNotNone(MqttDccComms(edge_system_name=self.edge_system.name,
+                                          url=self.url, port=self.port,
+                                          client_id="unit-test", clean_session=self.clean_session))
+
+    def test_init_with_msgattr(self):
+        """
+        Test MqttDccComms initialisation with:
+        #1. Valid messaging attributes
+        #2. Invalid messaging attributes
+        :return:
+        """
+        Mqtt.connect_soc = mock_connect_soc
+
+        # Test with valid messaging attributes
+        self.assertIsNotNone(MqttDccComms(edge_system_name=self.edge_system.name,
+                                          url=self.url, port=self.port, clean_session=self.clean_session,
+                                          mqtt_msg_attr=self.mqtt_msg_attr))
+
+        # Test with invalid messaging attributes
+        self.assertRaises(TypeError, lambda: MqttDccComms(edge_system_name=self.edge_system.name,
+                                                          url=self.url, port=self.port,
+                                                          clean_session=self.clean_session, mqtt_msg_attr=""))
+
+    def test_disconnect(self):
+        """
+        Test MqttDccComms _disconnect method
+        :return:
+        """
+        # Monkey patched MQTT disconnect method
+        Mqtt.disconnect = mock_disconnect
+
+        self.assertIsNone(self.client._disconnect())
+
+    def test_publish(self):
```

Comment:
Rename test case to test_send().  There is no publish method in dcccomms.  Please Rewrite this test case.  We should test the functionality rather than return type. 
- Create a MessagingAttributes object and pass it to publish().  Make sure that the underlying mocked publish() of transports is called with topic, message, qos, callback
- Dont pass any messagingAttribute object.  Now, publish should pick-up auto-generated topic, qos and callback.  Make sure that the underlying mocked publish() of transports is called with topic, message, qos and callback